src/api/modules/auth/Guards/roleguard.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+  constructor(private reflector: Reflector) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const roles = this.reflector.get<string[]>('roles', context.getHandler());
+    if (!roles) {
+      return true;
+    }
+    const request = context.switchToHttp().getRequest();
+      const user = request.user;
+      
+      const hasrole = roles.indexOf(user.role) > -1 
+      if (hasrole) { 
```

Comment:
there is no use for the hasRole variable, just return the logic directly

`return roles.indexOf(user.role) > -1;`


It is important that you use `;` at the end of your lines, learn it incase you go to other languages that are strict with it, you wont struggle with it